subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestInputAnnotationFailuresIntegrationTest.groovy

```diff
@@ -0,0 +1,222 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class TestInputAnnotationFailuresIntegrationTest extends AbstractIntegrationSpec {
+    def "using @Input annotation on #fieldType file fields fails validation with helpful error message"() {
+        given:
+        buildFile << """
+            import groovy.transform.CompileStatic
+
+            plugins {
+                id 'java'
+            }
+
+            ${mavenCentralRepository()}
+
+            @CacheableTask
+            @CompileStatic
+            abstract class MyTask extends DefaultTask {
+                @Input
+                $fieldInitialization
+
+                @TaskAction
+                void action() {
+                    logger.warn("Input file: {}", $fieldRead)
+                }
+            }
+
+            tasks.register('myTask', MyTask)
+        """
+
+        expect:
+        fails 'myTask'
+        result.assertHasErrorOutput("A problem was found with the configuration of task ':myTask' (type 'MyTask').")
+        result.assertHasErrorOutput("- Type 'MyTask' property 'myField' has @Input annotation used on property of type '$fieldType'.")
+        result.assertHasErrorOutput("Reason: A property of type '$fieldType' annotated with @Input cannot determine how to interpret the file.")
+        result.assertHasErrorOutput("Possible solutions:")
+        result.assertHasErrorOutput("1. Annotate with @InputFile for regular files.")
+        result.assertHasErrorOutput("2. Annotate with @InputFiles for collections of files.")
+        result.assertHasErrorOutput(". If you want to track the path, return File.absolutePath as a String and keep @Input.")
+
+        where:
+        fieldType       | fieldInitialization                                                               | fieldRead
+        'File'          | "File myField = project.layout.projectDirectory.file('myFile.txt').getAsFile()"   | 'myField.absolutePath'
+        'RegularFile'   | "RegularFile myField = project.layout.projectDirectory.file('myFile.txt')"        | 'myField.getAsFile().absolutePath'
+    }
+
+    def "using @Input annotation on #fieldType directory fields fails validation with helpful error message"() {
+        given:
+        buildFile << """
+            import groovy.transform.CompileStatic
+
+            plugins {
+                id 'java'
+            }
+
+            ${mavenCentralRepository()}
+
+            @CacheableTask
+            @CompileStatic
```

Comment:
In those tests, why do you have the `@CompileStatic` and why the `@CacheableTask`. I think the errors should be reported even if the task is not cacheable, right?

Reply:
I think this is leftover from my initial investigations trying to reproduce this problem.  Looks like they don't make a different.  I will remove.